presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileReader.java

```diff
@@ -286,11 +286,15 @@ public void close()
         rowGroupPosition = 0;
         rowGroupRowCount = 0;
         currentChunkRowCount = 0;
-        input.close();
-        if (decompressor != null) {
-            decompressor.destroy();
-        }
 
+        try {
+            input.close();
+        }
+        finally {
+            if (decompressor != null) {
+                decompressor.destroy();
```

Comment:
:D when I now read my previous comment, it looks as if I was suggesting to actually swallow `input.close()`'s exception (and not to avoid this). Natural language is hard :)